### ISSUE TITLE
Solution 3 of Nested Views

### ISSUE THREAD

**Issue by baoshan** [addyosmani/backbone-fundamentals#448]
Since `Backbone.View` relies on `jQuery.on`, which in turn bind the event to `el`, not the element who fires the event, why `delegateEvents` is still required after `$el.html(template)`?

**Comment by jridgewell** [comment]
It's because of OuterView's `render` calling `this.$el.html(template);`. jQuery's `html` remove all nested DOM elements and unbinds all of events on those elements. Meaning, when OuterView clears it's `$el`, it also clears (and unbinds) InnerView's `$el`.

A more appropriate solution could be:

``` javascript
var OuterView = Backbone.View.extend({
    initialize: function() {
        this.inner = new InnerView();
    },

    render: function() {
        this.inner.$el.detach(); // Detach InnerView before reseting OuterView's $el

        this.$el.html(template); // or this.$el.empty() if you have no template

        this.$el.append(this.inner.$el);
    }
});

var InnerView = Backbone.View.extend({
    render: function() {
        this.$el.html(template);
    }
});
```

Since we `detach` InnerView before we reset OuterView's `$el`, we save it from being cleaned (and unbinded). Afterwards, we just need to append InnerView again, saving the expensive `delegateEvents` call.
